sdk/storage/azblob/internal/shared/shared.go

```diff
@@ -98,9 +98,18 @@ func ParseConnectionString(connectionString string) (ParsedConnectionString, err
 		if !ok {
 			return ParsedConnectionString{}, errors.New("connection string missing AccountKey and SharedAccessSignature")
 		}
-		return ParsedConnectionString{
-			ServiceURL: fmt.Sprintf("%v://%v.blob.%v/?%v", defaultScheme, accountName, defaultSuffix, sharedAccessSignature),
-		}, nil
+
+		blobEndpoint, ok := connStrMap["BlobEndpoint"]
```

Comment:
I created the connection string from SAS using the portal. Its format is like this,

`BlobEndpoint=https://xxxxxxxx.blob.core.windows.net/;QueueEndpoint=https://xxxxxxxxx.queue.core.windows.net/;FileEndpoint=https://xxxxxxxx.file.core.windows.net/;TableEndpoint=https://xxxxxxxx.table.core.windows.net/;SharedAccessSignature=xxxxxxxxxxxxxxxx`

There is no `AccountName` here. So, the above code for checking account name also needs a fix.
```
accountName, ok := connStrMap["AccountName"]
if !ok {
	return ParsedConnectionString{}, errors.New("connection string missing AccountName")
}
```

Reply:
If you want, we can update this logic and fix this in some other PR.
I have done a similar fix in `azfile` package.
https://github.com/Azure/azure-sdk-for-go/pull/20798

Reply:
Hi @souravgupta-msft, you're right.

I've added further tests, and then refactored the ParseConnectionString because the logic was too confusing and duplicated.

Reply:
Resolving thread as code is now outdated